src/lv_core/lv_refr.c

```diff
@@ -741,6 +741,13 @@ static void lv_refr_vdb_flush(void)
 
     /*Flush the rendered content to the display*/
     lv_disp_t * disp = _lv_refr_get_disp_refreshing();
+#if LV_USE_GPU
```

Comment:
It's a little bit misleading (will be updated) but `LV_USE_GPU` enables only the 2 GPU callbacks in the display driver. Independently DMA2D (ST) os PXP (NXP) GPU can be enabled.

So these `#if LV_USE_GPU` statements are not required.

Reply:
Do you mean DMA2D might be used even though `LV_USE_GPU` is disabled? Our solution implements two gpu callbacks to accelerate rendering using the GPU just like DMA2D.

Reply:
> Do you mean DMA2D might be used even though LV_USE_GPU is disabled?

Yes, the GPU callbacks in the display driver have priority. If they are set LVGL uses them. If not set LVGL tries DMA2D, if it's not present or can't be used in the a given case, software rendering is applied. 


> Our solution implements two gpu callbacks to accelerate rendering using the GPU just like DMA2D.

You can simply remove the `#if LV_USE_GPU` guards as the `gpu_wait_cb` is useful for DMA2D too.

---

Probably `LV_USE_GPU` should be removed from LVGL as it causes confusion and saves only a very little RAM.